app/views/RoomInfoEditView/index.js

```diff
@@ -284,6 +292,72 @@ class RoomInfoEditView extends React.Component {
 		}, 100);
 	}
 
+	handleDeleteTeam = async(selected) => {
+		const { navigation, isMasterDetail } = this.props;
+		const { room } = this.state;
+		try {
+			const result = await RocketChat.deleteTeam({ teamId: room.teamId, ...(selected && { roomsToRemove: selected }) });
+			if (result.success) {
+				if (isMasterDetail) {
+					navigation.navigate('DrawerNavigator');
+				} else {
+					navigation.navigate('RoomsListView');
+				}
+			}
+		} catch (e) {
+			log(e);
+			showErrorAlert(
+				e.data.error
+					? I18n.t(e.data.error)
+					: I18n.t('There_was_an_error_while_action', { action: I18n.t('deleting_team') }),
+				I18n.t('Cannot_leave')
+			);
+		}
+	}
+
+	deleteTeam = async() => {
+		const { room } = this.state;
+		const { navigation } = this.props;
+
+		try {
+			const db = database.active;
+			const subCollection = db.get('subscriptions');
+			const teamChannels = await subCollection.query(
+				Q.where('team_id', room.teamId),
+				Q.where('team_main', null)
+			);
+
+			if (teamChannels.length) {
+				navigation.navigate('SelectListView', {
+					title: 'Delete_Team',
+					data: teamChannels,
+					infoText: 'Select_channels_to_delete',
+					nextAction: () => {
+						showConfirmationAlert({
+							title: I18n.t('Confirmation'),
+							message: I18n.t('You_are_deleting_the_team', { team: RocketChat.getRoomTitle(room) }),
+							confirmationText: I18n.t('Yes_action_it', { action: I18n.t('delete') }),
+							onPress: data => this.handleDeleteTeam(data)
+						});
+					}
+				});
+			} else {
+				showConfirmationAlert({
+					title: I18n.t('Confirmation'),
+					message: I18n.t('You_are_deleting_the_team', { team: RocketChat.getRoomTitle(room) }),
+					confirmationText: I18n.t('Yes_action_it', { action: I18n.t('delete') }),
+					onPress: () => this.handleDeleteTeam()
+				});
+			}
+		} catch (e) {
+			showConfirmationAlert({
```

Comment:
Can you explain why we're calling this here?
If `this.handleDeleteTeam` throws, is this going to run again?

Reply:
This is wrong, fixed